tests/test_instance.py

```diff
@@ -1,27 +1,46 @@
 import unittest
-from pynio import Instance
+from pynio import Instance, Block, Service
 from unittest.mock import MagicMock, patch
 from .mock import mock_service
 
 
+class MockInstance(Instance):
+
+    def __init__(self, host='127.0.0.1', port=8181, creds=None):
+        self._get_blocks = MagicMock()
+        self._get_blocks.return_value = {}, {}
+        self._get_services = MagicMock()
+        self._get_services.return_value = {}
+        super().__init__(host, port, creds)
+
+
 class TestInstance(unittest.TestCase):
-    def setUp(self):
-        blks = patch.object(Instance, '_get_blocks')
-        blks.start()
-        Instance._get_blocks.return_value = None, None
-        servs = patch.object(Instance, '_get_services')
-        servs.start()
 
     def test_instance(self):
-        i = Instance()
+        i = MockInstance()
         self.assertEqual(i.host, '127.0.0.1')
         self.assertEqual(i.port, 8181)
 
-    # TODO: Create a test that loads system settings
+    def test_add_block(self):
+        i = MockInstance()
+        i.blocks_types = MagicMock()
+        i._put = MagicMock()
+        name = 'name'
+        self.assertTrue(name not in i.blocks)
+        i.add_block(Block(name, 'type'))
+        self.assertTrue(name in i.blocks)
+
+    def test_add_service(self):
+        i = MockInstance()
+        i._put = MagicMock()
+        name = 'name'
+        self.assertTrue(name not in i.services)
+        i.add_service(Service(name))
+        self.assertTrue(name in i.services)
 
     def test_delete_all(self, *args):
         names = ['one', 'two', 'three']
-        i = Instance()
+        i = MockInstance()
```

Comment:
I think it's simple enough to keep it in here for now. No need to complicate things by splitting up into multiple files if we don't have to.